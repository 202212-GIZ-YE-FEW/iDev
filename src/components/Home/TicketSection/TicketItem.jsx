import { withTranslation } from "next-i18next";

import Button from "@/components/ui/Button";
function TicketItem({ t, numberOfTickets, price }) {
    return (
        <>
            <div className='bg-light-white flex flex-col items-center justify-center text-center py-16 rounded-3xl space-y-6 drop-shadow-lg'>
                <span className='uppercase text-4xl'>
                    {numberOfTickets} {t("ticket")}
                </span>
                <span className='text-base rtl:text-3xl text-black/50'>
                    {price}$
                </span>
                <Button
                    content={t("purchase")}
                    textTransform='uppercase'
                    filled='true'
                    size='large'
                    fontSize='text-lg md:text-xl lg:text-2xl'
                    radius='md'
                />
            </div>
        </>
    );
}
export default withTranslation("home")(TicketItem);
