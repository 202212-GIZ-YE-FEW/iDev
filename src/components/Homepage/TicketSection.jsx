import { withTranslation } from "next-i18next";
import { useQuery } from "react-query";

import "react-multi-carousel/lib/styles.css";

import getDocument from "@/firebase/getData";

import AreYouCounselor from "./AreYouCounselor";
import TicketItem from "./TicketItem";
import PageIntro from "../PageIntro";

function TicketSection({ t }) {
    const {
        isLoading,
        error,
        data: result,
    } = useQuery("tickets", async () => {
        const data = await getDocument("tickets");
        return data;
    });

    return (
        <>
            <div className='container flex flex-col justify-between'>
                {isLoading ? (
                    <div className='text-center'>
                        <div
                            className='inline-block h-8 w-8 animate-spin rounded-full border-4 border-solid border-current border-r-transparent align-[-0.125em] text-cyan motion-reduce:animate-[spin_1.5s_linear_infinite]'
                            role='status'
                        ></div>
                    </div>
                ) : (
                    <>
                        {result?.length > 0 ? (
                            <>
                                <PageIntro
                                    title={t("purchaseTicketsTitle")}
                                    subtitle={t("purchaseTicketsSubtitle")}
                                />
                                <div className='grid sm:grid-cols-1 md:grid-cols-2 my-10 lg:grid-cols-3 gap-7'>
                                    {result?.map((item, index) => {
                                        return (
                                            <TicketItem
                                                key={index}
                                                ticketID={item.id}
                                                numberOfTickets={
                                                    item.number_of_tickets
                                                }
                                                price={item.price}
                                            />
                                        );
                                    })}
                                </div>
                            </>
                        ) : (
                            <></>
                        )}
                    </>
                )}
                <AreYouCounselor />
            </div>
        </>
    );
}
export default withTranslation("home")(TicketSection);
