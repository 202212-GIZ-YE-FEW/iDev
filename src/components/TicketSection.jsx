import { withTranslation } from "next-i18next";

import "react-multi-carousel/lib/styles.css";

import AreYouCounselor from "./AreYouCounselor";
import PageIntro from "./PageIntro";
import TicketItem from "./TicketItem";
function TicketSection({ t }) {
    const ticketPlans = [
        {
            numberOfTickets: 5,
            price: 10,
        },
        {
            numberOfTickets: 25,
            price: 40,
        },
        {
            numberOfTickets: 50,
            price: 70,
        },
    ];
    return (
        <>
            <div className='container flex flex-col justify-between'>
                <PageIntro
                    title={t("purchaseTicketsTitle")}
                    subtitle={t("purchaseTicketsSubtitle")}
                />
                <div className='grid sm:grid-cols-1 md:grid-cols-2 my-10 lg:grid-cols-3 gap-7'>
                    {ticketPlans.map((item, index) => {
                        return (
                            <TicketItem
                                key={index}
                                numberOfTickets={item.numberOfTickets}
                                price={item.price}
                            />
                        );
                    })}
                </div>
                <AreYouCounselor />
            </div>
        </>
    );
}
export default withTranslation("home")(TicketSection);
