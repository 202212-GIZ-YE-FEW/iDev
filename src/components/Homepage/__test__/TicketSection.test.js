import { QueryClient, QueryClientProvider } from "react-query";
import renderer from "react-test-renderer";

import TicketSection from "../TicketSection";

const queryClient = new QueryClient();

it("renders correctly", () => {
    const tree = renderer
        .create(
            <QueryClientProvider client={queryClient}>
                <TicketSection />
            </QueryClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
