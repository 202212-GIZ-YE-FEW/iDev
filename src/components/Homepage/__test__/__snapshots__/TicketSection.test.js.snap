// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container flex flex-col justify-between"
>
  <div
    className="mb-5 text-start"
  >
    <span
      className="font-normal block text-3xl md:text-4xl rtl:md:text-3xl lg:text-5xl rtl:lg:text-4xl uppercase break-words"
    >
      purchaseTicketsTitle
    </span>
    <p
      className="text-lg md:text-xl lg:text-2xl text-black/50"
    >
      purchaseTicketsSubtitle
    </p>
  </div>
  <div
    className="grid sm:grid-cols-1 md:grid-cols-2 my-10 lg:grid-cols-3 gap-7"
  >
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center py-16 rounded-3xl space-y-6 drop-shadow-lg"
    >
      <span
        className="uppercase text-4xl"
      >
        5
         
        ticket
      </span>
      <span
        className="text-base rtl:text-3xl text-black/50"
      >
        10
        $
      </span>
      <div
        className="w-fit"
      >
        <button
          className="font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
          disabled={false}
        >
          purchase
        </button>
      </div>
    </div>
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center py-16 rounded-3xl space-y-6 drop-shadow-lg"
    >
      <span
        className="uppercase text-4xl"
      >
        25
         
        ticket
      </span>
      <span
        className="text-base rtl:text-3xl text-black/50"
      >
        40
        $
      </span>
      <div
        className="w-fit"
      >
        <button
          className="font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
          disabled={false}
        >
          purchase
        </button>
      </div>
    </div>
    <div
      className="bg-light-white flex flex-col items-center justify-center text-center py-16 rounded-3xl space-y-6 drop-shadow-lg"
    >
      <span
        className="uppercase text-4xl"
      >
        50
         
        ticket
      </span>
      <span
        className="text-base rtl:text-3xl text-black/50"
      >
        70
        $
      </span>
      <div
        className="w-fit"
      >
        <button
          className="font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
          disabled={false}
        >
          purchase
        </button>
      </div>
    </div>
  </div>
  <div
    className="bg-light-white flex flex-col items-center justify-center text-center py-10 px-5 rounded-md space-y-6 drop-shadow-lg"
  >
    <span
      className="uppercase text-4xl"
    >
      askIfCounselor
    </span>
    <span
      className="text-base rtl:text-xl text-black"
    >
      askCounselor
    </span>
    <div>
      <a
        href="/requirements"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="font-normal whitespace-nowrap cursor-pointer  uppercase text-lg md:text-xl lg:text-2xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
          disabled={false}
        >
          learnMore
        </button>
      </a>
    </div>
  </div>
</div>
`;
